### ISSUE TITLE
Erroneous L3 miss ratio on Sandy Bridge

### ISSUE THREAD

**Issue by LangdalP** [RRZE-HPC/likwid#83]
I recently noticed that the L3 miss ratio must be wrongly calculated, as it does not range between 0.0-1.0 but 0 and some very high number. Specifically, it is equal to the `OFFCORE_RESPONSE_0_OPTIONS STAT` event.

Here is an example output:
```
--------------------------------------------------------------------------------
CPU name:       Intel(R) Core(TM) i7-2600K CPU @ 3.40GHz
CPU type:       Intel Core SandyBridge processor
CPU clock:      3.41 GHz
--------------------------------------------------------------------------------
Finding primes up to 30000000 on 4 threads...
Prime-hunting completed in 9.76061 seconds!
--------------------------------------------------------------------------------
Region thread-primes, Group 1: L3CACHE
+-------------------+----------+----------+----------+----------+
|    Region Info    |  Core 0  |  Core 1  |  Core 2  |  Core 3  |
+-------------------+----------+----------+----------+----------+
| RDTSC Runtime [s] | 8.701322 | 8.703926 | 8.701994 | 8.698831 |
|     call count    |        1 |        1 |        1 |        1 |
+-------------------+----------+----------+----------+----------+

+----------------------------+------------------------------------+-------------+-------------+-------------+-------------+
|            Event           |               Counter              |    Core 0   |    Core 1   |    Core 2   |    Core 3   |
+----------------------------+------------------------------------+-------------+-------------+-------------+-------------+
|      INSTR_RETIRED_ANY     |                FIXC0               | 16862930000 | 16855270000 | 16792930000 | 16830350000 |
|    CPU_CLK_UNHALTED_CORE   |                FIXC1               | 29634070000 | 29640220000 | 29570710000 | 29610950000 |
|    CPU_CLK_UNHALTED_REF    |                FIXC2               | 29642730000 | 29647960000 | 29578140000 | 29618020000 |
| OFFCORE_RESPONSE_0_OPTIONS | PMC0:MATCH0=0x0081:MATCH1=0x3fffc0 |       59622 |      132006 |      131938 |       46209 |
| OFFCORE_RESPONSE_1_OPTIONS |    PMC1:MATCH0=0x0081:MATCH1=0x1   |      151020 |      768132 |      636170 |      111839 |
|       L1D_REPLACEMENT      |                PMC2                |      202008 |     1516084 |     1004697 |      180602 |
+----------------------------+------------------------------------+-------------+-------------+-------------+-------------+

+---------------------------------+------------------------------------+--------------+-------------+-------------+-------------+
|              Event              |               Counter              |      Sum     |     Min     |     Max     |     Avg     |
+---------------------------------+------------------------------------+--------------+-------------+-------------+-------------+
|      INSTR_RETIRED_ANY STAT     |                FIXC0               |  67341480000 | 16792930000 | 16862930000 | 16835370000 |
|    CPU_CLK_UNHALTED_CORE STAT   |                FIXC1               | 118455950000 | 29570710000 | 29640220000 | 29613987500 |
|    CPU_CLK_UNHALTED_REF STAT    |                FIXC2               | 118486850000 | 29578140000 | 29647960000 | 29621712500 |
| OFFCORE_RESPONSE_0_OPTIONS STAT | PMC0:MATCH0=0x0081:MATCH1=0x3fffc0 |       369775 |       46209 |      132006 |  92443.7500 |
| OFFCORE_RESPONSE_1_OPTIONS STAT |    PMC1:MATCH0=0x0081:MATCH1=0x1   |      1667161 |      111839 |      768132 | 416790.2500 |
|       L1D_REPLACEMENT STAT      |                PMC2                |      2903391 |      180602 |     1516084 | 725847.7500 |
+---------------------------------+------------------------------------+--------------+-------------+-------------+-------------+

+----------------------+--------------+--------------+--------------+--------------+
|        Metric        |    Core 0    |    Core 1    |    Core 2    |    Core 3    |
+----------------------+--------------+--------------+--------------+--------------+
|  Runtime (RDTSC) [s] |       8.7013 |       8.7039 |       8.7020 |       8.6988 |
| Runtime unhalted [s] |       8.6877 |       8.6895 |       8.6691 |       8.6809 |
|      Clock [MHz]     |    3410.0456 |    3410.1516 |    3410.1853 |    3410.2279 |
|          CPI         |       1.7573 |       1.7585 |       1.7609 |       1.7594 |
|    L3 request rate   | 8.955739e-06 | 4.557222e-05 | 3.788320e-05 | 6.645079e-06 |
|     L3 miss rate     | 3.535684e-06 | 7.831735e-06 | 7.856759e-06 | 2.745576e-06 |
|     L3 miss ratio    |        59622 |       132006 |       131938 |        46209 |
+----------------------+--------------+--------------+--------------+--------------+

+---------------------------+--------------+--------------+--------------+--------------+
|           Metric          |      Sum     |      Min     |      Max     |      Avg     |
+---------------------------+--------------+--------------+--------------+--------------+
|  Runtime (RDTSC) [s] STAT |      34.8060 |       8.6988 |       8.7039 |       8.7015 |
| Runtime unhalted [s] STAT |      34.7272 |       8.6691 |       8.6895 |       8.6818 |
|      Clock [MHz] STAT     |   13640.6104 |    3410.0456 |    3410.2279 |    3410.1526 |
|          CPI STAT         |       7.0361 |       1.7573 |       1.7609 |       1.7590 |
|    L3 request rate STAT   |       0.0001 | 6.645079e-06 | 4.557222e-05 | 2.476406e-05 |
|     L3 miss rate STAT     | 2.196975e-05 | 2.745576e-06 | 7.856759e-06 | 5.492439e-06 |
|     L3 miss ratio STAT    |       369775 |        46209 |       132006 |   92443.7500 |
+---------------------------+--------------+--------------+--------------+--------------+
```
I inspected the file file `likwid/groups/sandybridge/L3CACHE.txt` and observed that that `OFFCORE_RESPONSE_0_OPTIONS` was defined slightly differently in the EVENTSET section compared to how it appears as a divisor in the metric calculation. I changed the file to fix this, and it appears to be fixed now.

The suggested changes can be seen in pull request #82 .

**Comment by LangdalP** [comment]
Can be closed as PR #82 has been merged.

**Comment by TomTheBear** [comment]
OK. Thanks
